src/op_analytics/datapipeline/models/templates/erc20_contracts/erc20_contract_creation.sql.j2

```diff
@@ -0,0 +1,14 @@
+SELECT
+  ct.*
+  , 'erc20' AS token_standard
+  , sc.token_type
+FROM {{ contract_creation_traces }} AS ct
+LEFT JOIN {{ superchainerc20 }} AS sc
+  ON
+    ct.chain_id = sc.chain_id
+    AND ct.contract_address = sc.contract_address
+    AND ct.network = sc.network
+WHERE
+  -- transfer(address,uint256): 0xa9059cbb
```

Comment:
Nice approach, some things to check on:

1. Does this catch tokens created behind a proxy (like USDC https://optimistic.etherscan.io/token/0x0b2c639c533813f4aa9d7837caf62653d097ff85#readContract)? This is useful for the underlying contracts at least, but we may need to use transfers to get the tokens themselves.

^ It seems like you get this in your approach by calling this 'erc20_contract_creation'. So this isn't the tokens table (yet) but it could be a contributor.

2. Are there other functions in the [ERC20 standard](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#IERC20) that we want to check for? Like approve()

We can ask protocol/eng if this approach is sensible as well, or if there's nuance to look out for (like do we need PUSH0 prepending, or something else)

Reply:
We'll also need to load in a static/manual list of pre-regenesis tokens that won't have a creation trace on OPM.

We have that somewhere.

Reply:
I think the issue with the current pipeline is that it's by blockbatch, so technically the logic won't be able to look back at the entire history (this is where dbt could come in) 
- given the constraint of using the bytecode to identify erc20 in the above example, and if we want to use joining `creation_traces` + `erc20_transfers`
- that means we need to create erc20 token first transfers with contract_address and it's min(block_timestamp) and then join to contract creation to see if there's a match

cc: @lithium323 if the above understanding is correct